cmd/admin-rpc-server.go

```diff
@@ -158,6 +162,84 @@ func (s *adminCmd) GetConfig(args *AuthRPCArgs, reply *ConfigReply) error {
 	return nil
 }
 
+// WriteConfigArgs - wraps the bytes to be written and temporary file name.
+type WriteConfigArgs struct {
+	AuthRPCArgs
+	TmpFileName string
+	Buf         []byte
+}
+
+// WriteConfigReply - wraps the result of a writing config into a temporary file.
+// the remote node.
+type WriteConfigReply struct {
+	AuthRPCReply
+	N int64
+}
+
+// WriteTmpConfig - writes the supplied config contents onto the
+// supplied temporary file.
+func (s *adminCmd) WriteTmpConfig(wArgs *WriteConfigArgs, wReply *WriteConfigReply) error {
+	if err := wArgs.IsAuthenticated(); err != nil {
+		return err
+	}
+
+	configDir, err := getConfigPath()
+	if err != nil {
+		errorIf(err, "Failed to get config path")
+		return err
+	}
+
+	// Create temporary config file at configDir/TmpFileName.
+	tmpFile, err := os.Create(filepath.Join(configDir, wArgs.TmpFileName))
+	if err != nil {
+		errorIf(err, "Failed to create temporary config file.")
+		return err
+	}
```

Comment:
Just use `ioutil.WriteFile` , also looks like a duplicate code to me.. `WriteTmpConfig()` can be re-purposed here.